tensorboard/webapp/app_routing/views/router_link_directive_container.ts

```diff
@@ -47,9 +55,12 @@ export class RouterLinkDirectiveContainer {
   get href() {
     if (!this.pathname) return null;
 
-    return this.location.getFullPathFromRouteOrNav({
-      pathname: this.pathname,
-    });
+    return (
+      this.appRoot.slice(0, -1) +
```

Comment:
Would it maybe make sense to expose `getAbsPathnameWithAppRoot()` somewhere to kind of share this logic?  As it is, it's not terribly obvious here why we slice the last char off appRoot unless you can see the bigger picture.  Maybe at least a comment would help.

Reply:
Done. Moved it to AppRootProvider instead.